### PR TITLE
Create DEVELOPER_GUIDE.md

### PR DESCRIPTION
**Pull request by joshpalis**
basic developer guide

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+
```
Comment (owaiskazi19) [inline DEVELOPER_GUIDE.md:1]: Can you write some intro about the SDK?
Ref: https://github.com/opensearch-project/OpenSearch/issues/1422

**Review by owaiskazi19 — COMMENTED** [review]
Index needed for quick access of the sections

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (joshpalis) [reply DEVELOPER_GUIDE.md:1]: sure, ill also include links to that issue and also the pluggable architecture blog post as well

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+- [OpenSearch SDK Developer Guide](#opensearch-sdk-developer-guide)
+  - [Introduction](#introduction)
+  - [Getting Started](#getting-started)
+    - [Git Clone IndependentPlugin Repo](#git-clone-independentplugin-repo)
+	- [Run Independent Plugin](#run-independent-plugin)
+	- [Run Tests](#run-tests)
+	- [Send Message using Telnet](#send-message-using-telnet)
+
+# OpenSearch SDK Developer Guide
+
```
Comment (owaiskazi19) [inline DEVELOPER_GUIDE.md]: Can you move the index here? Also, you can remove `OpenSearch SDK Developer Guide` from the index.

**Review by owaiskazi19 — APPROVED** [review]
LGTM!

**Pull request merged**